lib/release.js

```diff
@@ -0,0 +1,437 @@
+
+    const upstreamHref = runSync('git', [
+      'config', '--get',
+      `remote.${upstream}.url`]).trim();
+    if (!new RegExp(`${owner}/${repo}(?:.git)?$`).test(upstreamHref)) {
+      cli.warn('Remote repository URL does not point to the expected ' +
+        `repository ${owner}/${repo}`);
+    }
+  }
+
+  async prepare() {
+    const { cli, newVersion, versionComponents } = this;
+
+    // Create new proposal branch.
+    const shouldBranch = await cli.prompt(
+      `Create new proposal branch for ${newVersion}?`, true);
+    if (!shouldBranch) return this.abort();
+    await this.createProposalBranch();
+
+    // Update version and release info in src/node_version.h.
+    const shouldUpdateNodeVersion = await cli.prompt(
+      `Update 'src/node_version.h' for ${newVersion}?`, true);
+    if (!shouldUpdateNodeVersion) return this.abort();
+    await this.updateNodeVersion();
+
+    // Check whether to update NODE_MODULE_VERSION (default false).
+    const shouldUpdateNodeModuleVersion = await cli.prompt(
+      'Update NODE_MODULE_VERSION?', false);
+    if (shouldUpdateNodeModuleVersion) {
+      const runtime = await cli.promptInput(
+        'Specify runtime (ex. \'node\') for new NODE_MODULE_VERSION:',
+        false);
+      const variant = await cli.promptInput(
+        'Specify variant (ex. \'v8_7.9\') for new NODE_MODULE_VERSION:',
+        false);
+      const versions = await cli.promptInput(
+        'Specify versions (ex. \'14.0.0-pre\') for new NODE_MODULE_VERSION:',
+        false);
+      this.updateNodeModuleVersion(runtime, variant, versions);
+    }
+
+    // Update any REPLACEME tags in the docs.
+    const shouldUpdateREPLACEMEs = await cli.prompt(
+      'Update REPLACEME items in docs?', true);
+    if (!shouldUpdateREPLACEMEs) return this.abort();
+    await this.updateREPLACEMEs();
+
+    this.date = await cli.promptInput(
+      'Enter release date in YYYY-MM-DD format:');
+
+    // Update Changelogs
+    const shouldUpdateChangelogs = await cli.prompt(
+      'Update changelogs?', true);
+    if (!shouldUpdateChangelogs) return this.abort();
+
+    cli.startSpinner('Updating CHANGELOG.md');
+    await this.updateMainChangelog();
+    cli.stopSpinner('Updated CHANGELOG.md');
+
+    cli.startSpinner(`Updating CHANGELOG_V${versionComponents.major}.md`);
+    await this.updateMajorChangelog();
+    cli.stopSpinner(`Updated CHANGELOG_V${versionComponents.major}.md`);
+
+    await cli.prompt('Finished editing the changelogs?', false);
+
+    // Create release commit.
+    const shouldCreateReleaseCommit = await cli.prompt(
+      'Create release commit?', true);
+    if (!shouldCreateReleaseCommit) return this.abort();
+
+    // Proceed with release only after the releaser has amended
+    // it to their liking.
+    const created = await this.createReleaseCommit();
+    if (!created) {
+      const lastCommitSha = runSync('git', ['rev-parse', '--short', 'HEAD']);
```

Comment:
what is the case where we would hit this?

Reply:
i went with a similar flow as git node land - sometimes you don't want the amended commit message and instead want to edit it yourself to add something that's notable but not tagged as such. In that case, i'd want to wait for the reviewer to give me the go-ahead (& allow them to make those edits) before continuing, i think